closure/testing/closure_js_test.bzl

```diff
@@ -14,128 +14,56 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Build rule for running Closure Library JsUnit tests in PhantomJS.
-"""
+"""Build macro for running Closure Library JsUnit tests in PhantomJS."""
 
 # XXX: It would be significantly faster and produce better stacktraces if we
 #      could avoid compilation by running in raw sources mode. This is not
 #      possible due to a resource loading bug in PhantomJS.
 #      https://github.com/ariya/phantomjs/issues/14028
 
-load("//closure/private:defs.bzl",
-     "CLOSURE_LIBRARY_BASE_ATTR",
-     "CLOSURE_LIBRARY_DEPS_ATTR",
-     "JS_DEPS_ATTR",
-     "JS_HIDE_WARNING_ARGS",
-     "JS_LANGUAGE_DEFAULT",
-     "JS_PEDANTIC_ARGS",
-     "JS_TEST_FILE_TYPE",
-     "collect_js_srcs",
-     "determine_js_language")
+load("//closure/compiler:closure_js_binary.bzl", "closure_js_binary")
+load("//closure/compiler:closure_js_library.bzl", "closure_js_library")
+load("//closure/private:defs.bzl", "JS_LANGUAGE_DEFAULT")
+load("//closure/testing:phantomjs_test.bzl", "phantomjs_test")
 
-# Maps input language to preferred output language. Since PhantomJS (based on
-# QtWebKit) supports ECMASCRIPT5_STRICT, we want to transpile to that language,
-# or whatever is nearest: http://kangax.github.io/compat-table/es5/#webkit
-_OUTPUT_LANGUAGE = {
-    "ECMASCRIPT3": "ECMASCRIPT3",
-    "ECMASCRIPT5": "ECMASCRIPT5",
-    "ECMASCRIPT6": "ECMASCRIPT5",
-    "ECMASCRIPT5_STRICT": "ECMASCRIPT5_STRICT",
-    "ECMASCRIPT6_STRICT": "ECMASCRIPT5_STRICT",
-    "ECMASCRIPT6_TYPED": "ECMASCRIPT5_STRICT",
-}
+def closure_js_test(name,
+                    srcs,
+                    deps = [],
+                    html=None,
+                    compilation_level="ADVANCED",
+                    css=None,
+                    defs=[],
+                    dependency_mode="LOOSE",
+                    entry_points=[],
+                    formatting="PRETTY_PRINT",
+                    language=JS_LANGUAGE_DEFAULT,
+                    pedantic=False,
+                    suppress=[]):
```

Comment:
For the defaults, I found the following section of the bazel [manual](http://bazel.io/docs/skylark/macros.html) interesting:

```
In most cases, optional parameters should have a default value of None. None
can be passed directly to native rules, which treat it the same as if you had not
passing any argument. Thus, there is no need to replace it with 0, False, or [] for
this purpose. Instead, the macro should defer to the rules it creates, as their
defaults may be complex or may change over time. Additionally, a parameter
that is explicitly set to its default value looks different than one that is never set
(or set to None) when accessed through the query language or build-system
internals.
```

What are your thoughts here?